src/devices/src/virtio/queue.rs

```diff
@@ -741,6 +765,91 @@ pub(crate) mod tests {
         assert_eq!(q.avail_event(m), 2);
     }
 
+    #[test]
+    #[should_panic(
+        expected = "The number of available virtio descriptors is greater than queue size!"
+    )]
+    fn test_invalid_avail_idx_no_notification() {
+        // This test ensures constructing a descriptor chain succeeds
+        // with valid available ring indexes while it produces an error with invalid
+        // indexes.
+        let m = &create_anon_guest_memory(&[(GuestAddress(0), 0x1000)], false).unwrap();
```

Comment:
doesn't really really matter for the test, but shouldn't the size here be `0x4000`?

Reply:
same in the other test.

Reply:
Uh actually it needs to be at least 0x4000 since we also need to reserve place for the contents of the queue, right?

Reply:
yes, it looks like it should be at least 0x5000 actually (4 descriptors plus the queue)